src/utils/streams/replace_stream.js

```diff
@@ -0,0 +1,46 @@
+import { Transform } from 'stream';
+
+export function createReplaceStream(toReplace, replacement) {
+  if (typeof toReplace !== 'string') {
+    throw new TypeError('toReplace must be a string');
+  }
+
+  let buffer = Buffer.alloc(0);
+  return new Transform({
+    objectMode: false,
+    async transform(value, enc, done) {
+      try {
+        buffer = Buffer.concat([buffer, value], buffer.length + value.length);
+
+        while (true) {
+          const index = buffer.indexOf(toReplace);
+          if (index === -1) {
+            break;
+          }
+
+          this.push(buffer.slice(0, index));
+          this.push(replacement);
+          buffer = buffer.slice(index + toReplace.length);
+        }
+
+        if (buffer.length > replacement.length) {
```

Comment:
__nit:__ could you please add comment explaining what this `if` case is for?

Reply:
Nice, this was actually a bug: https://github.com/elastic/kibana/pull/14007/commits/8ca1c33c232935c3e2e700e68414a121b43cac68